### ISSUE TITLE
adding my own script causes to delete everything on current folder!!!!!!!!

### ISSUE THREAD

**Issue by bosseyes** [PoE-TradeMacro/POE-TradeMacro#671]
AHK -64bit
Microsoft Windows [Version 10.0.16299.125]

I've just downloaded the newest release from this site: https://github.com/PoE-TradeMacro/POE-TradeMacro/releases

changed some settings here:
AdditionalMacros.ini
```
`; Converted with PoEScripts_ConvertAdditionalMacrosSettings() on 02-03-2018 / DO NOT EDIT OR REMOVE THIS COMMENT MANUALLY
[General]
KeyToSCState=0

KeyToSCState_Description="Short$ KeyToSC. Long$ "
EnableState=1
EnableState_Description="Short$ Enable AdditionalMacros. Long$ Enables or disables AdditionalMacros completely. `nNeeds a script reload to take effect."

[TogglePOEItemScript]
; Pause item parsing with the pause key (other macros remain).
Description=Pause item parsing with the pause key (other macros remain). (Default = Pause).
State=1
Hotkeys=Pause

[Minimize]
; Winkey+D minimizes the active PoE window (PoE stays minimized this way).
Description=Winkey+D minimizes the active PoE window (PoE stays minimized this way). (Default = WinKey + D).
State=0
Hotkeys=#D

[HighlightItems]
; Ctrl+F fills search bars in the stash or vendor screens with the item's name or info you're hovering over.
; Change behaviour:
;	Arg1 = Use broader terms, default: 0.
;	Arg2 = Leave the search field after pasting the search terms, default: 1.
Description=Fills search bars in the stash or vendor screens with the item's name or info you're hovering over. (Default = Ctrl + F).
State=0
Arg1=0
Arg2=1
Hotkeys=^F

[HighlightItemsAlt]
; Ctrl+Alt+F uses much broader search terms for the highlight function.
; Change behaviour:
;	Arg1 = Use broader terms, default: 1.
;	Arg2 = Leave the search field after pasting the search terms, default: 1.
Description=Uses much broader search terms for the highlight function. (Default = Ctrl+Alt+F).
State=0
Arg1=1
Arg2=1
Hotkeys=^!F

[LookUpAffixes]
; Opens poeaffix.net in your browser, navigating to the item that you're hovering over.
Description=Opens poeaffix.net in your browser, navigating to the item that you're hovering over. (Default = Ctrl + M).
State=0
Hotkeys=^M

[CloseScripts]
; Ctrl+Esc closes all running scripts specified by (and including) ItemInfo or TradeMacro.
Description=Closes all running scripts specified by (and including) ItemInfo or TradeMacro. (Default = Ctrl + Esc).
State=1
Hotkeys=ScrollLock

[KickYourself]
; Quickly leave a group by kicking yourself. Only works for one specific character name.
Description=Quickly leave a group by kicking yourself. Only works for one specific character name. (Default = F4).
State=1
Character=BitCooooonnect
Hotkeys=F9

[Hideout]
; Go to hideout with F5.
Description=Go to hideout from towns/other hideouts. (Default = F5).
State=0
Hotkeys=F5

[ScrollTabRight]
; Ctrl+scroll down or holding right mouse button+scroll down scrolls through stash tabs rightward.
Description=Scroll through stash tabs rightward. (Default = Ctrl+scroll down or holding right mouse button+scroll down).
State=1
Hotkeys=^WheelDown, ~RButton & WheelDown

[ScrollTabLeft]
; Ctrl+scroll up or holding right mouse button+scroll up scrolls through stash tabs leftward.
Description=Scroll through stash tabs leftward. (Default = Ctrl+scroll up or holding right mouse button+scroll up).
State=1
Hotkeys=^WheelUp, ~RButton & WheelUp

[SendCtrlC]
; Ctrl+right mouse button sends ctrl+c.
Description=Sends ctrl+c. (Default = Ctrl+right mouse button).
State=0
Hotkeys=^RButton

[Remaining]
; Mobs remaining with F9.
Description=Mobs remaining with F9. (Default = F9).
State=0
Hotkeys=F9

[JoinChannel]
; Join a channel with F10. Default: global 820.
Description=Join a channel. Default channel: global 820. (Default = F10).
State=0
Hotkeys=Numpad5
Channel=global 820

[SetAfkMessage]
; Pastes afk message to your chat and marks "X" so you can type in the estimated time.
Description=Pastes afk message to your chat and marks "X" so you can type in the estimated time. (Default = F11).
State=0
Hotkeys=Numpad6

[AdvancedItemInfo]
; Opens an item on pathof.info for an advanced affix breakdown.
Description=Opens an item on pathof.info for an advanced affix breakdown. (Default = Ctrl + Alt + C).
State=1
Hotkeys=^!c

[WhoisLastWhisper]
; Sends "/whois lastWhisperCharacterName" when keys are released (not when pressed).
Description=Sends '/whois lastWhisperCharacterName' when keys are released (not when pressed). (Default = Ctrl + Alt + Enter).
State=1
Hotkeys=^!Enter
`

I added my own code here:
customMacros_example.txt
`/*
	Custom macro main file. Don't delete it. Using and including this file is important. If you don't want to edit the 
	technical AdditionalMacro.txt file, you need this file and the already defined hotkey label. 
	Epsecially if you want to execute any personal code outside of hotkey labels.
	
	**TLDR**:
		You can add your own code below, use the existing Label.
		View a formatted description/how to here -> https://github.com/PoE-TradeMacro/POE-TradeMacro/wiki/Custom-Macros.

	*/

CM_ExecuteCustomMacrosCode_Label:	

;pause::pause
;ScrollLock::Suspend

F10::
SendInput, {Enter}
SendInput, ^{a}
SendInput, {Del}
SendInput, {Enter}
sleep, 70
SendInput, ^{Enter}
SendInput, busy atm, will answer you asap
SendInput, {Enter}
return

F11::
SendInput, {Enter}
SendInput, ^{a}
SendInput, {Del}
SendInput, {Enter}
sleep, 70
SendInput, ^{Enter}
SendInput, {Home}
SendInput, {Del}
SendInput, {/}invite{space}
SendInput, {Enter}
return

F12::
SendInput, {Enter}
SendInput, ^{a}
SendInput, {Del}
SendInput, {Enter}
sleep, 70
SendInput, ^{Enter}
SendInput, im doing uber lab, if you can wait, add me and whisper me when im done or i will whisper you then
sleep, 100
SendInput, {Enter}
return

XButton1::
SendInput, {Enter}
sleep, 20
SendInput, {/}hideout
SendInput, {Enter}
return

;LControl & RAlt::
;SendInput, {Enter}
;sleep, 20
;SendInput, {/}remaining
;SendInput, {Enter}
;return

XButton2::
SendInput, {Enter}
sleep, 20
SendInput, {/}oos
SendInput, {Enter}
return

NumpadDiv::
RepeatCount = 30
Loop, %RepeatCount%
{
	SendInput, {Enter}
	Random, rand1, 150, 250
	sleep, %rand1%
	SendInput, {/}trade %RepeatCount%
	Random, rand0, 1200, 1400
	sleep, %rand0%
	SendInput, {Enter}
	Random, rand2, 800, 1000
	sleep, %rand2%
	SendInput, {Enter}
	Random, rand3, 150, 250
	sleep, %rand3%
	SendInput, {Up}
	Random, rand4, 300, 400
	sleep, %rand4%
	SendInput, {Up}
	Random, rand5, 300, 400
	sleep, %rand5%
	SendInput, {Enter}
	Random, rand6, 150, 250
	sleep, %rand6%
	RepeatCount--
}
return

Return`
```
as soon as I start the script it deletes everything on my current folder/desktop and it is reproducable! JUST WTF HAVE YOU DONE? I have used an older version before in november last time i think, nothing was wrong with my scripts.

**Comment by aRTy42** [comment]
Do you have the script directly on your desktop, without using a folder? The only thing I can think of is that the updater tries to back up the script directory and if you don't have a proper folder for the script but instead have it directly on your desktop, it potentially tries to back up your whole desktop.

As far as I know there are checks and warning messages in place that tell you to avoid the desktop and even a folder on the desktop, because windows often causes permission issues there.

If you want to access the script from the desktop, use a link to the Run_TradeMacro.ahk file, but put the actual script folder in a location where programs are expected (such as C:\Program Files\PoE-TradeMacro\ )

**Comment by bosseyes** [comment]
no i have it in a folder, when i dont add my custom macros, everything is ok. but i dont see how my macros would possibly cause that chaos.

**Comment by Eruyome** [comment]
The updater shouldn't mess up anything here. Especially not without updating wich doesn't seem to be what has caused these issues, but starting the macro itself if I understand correctly. 

Are these issues caused by the unchanged script or only after adding your own macros?

**Comment by bosseyes** [comment]
only after adding my own macros in: customMacros_example.txt

**Comment by Eruyome** [comment]
Haven't checked the code yet but at a glance I'd say your code get's auto-executed at  script-start and code like this
```
SendInput, ^{a}
SendInput, {Del}
```
selects all files and deletes them.

**Comment by Eruyome** [comment]
I'll certainly have to take a look at this but for now you can try to add 
```
F12:
	IfWinActive, ahk_group PoEWindowGrp
	{
		;your code here
	}
return
```

**Comment by Eruyome** [comment]
I have overlooked something. You're adding your code into the `CM_ExecuteCustomMacrosCode_Label` Label, which is fine if you want it to be executed directly. But you don't want that. You want to add new hotkeys/labels, which you can't do inside another label.

This would be the right way:
```
CM_ExecuteCustomMacrosCode_Label:	
    ; keep empty
Return

F10::
	SendInput, {Enter}
	SendInput, ^{a}
	SendInput, {Del}
	SendInput, {Enter}
	sleep, 70
	SendInput, ^{Enter}
	SendInput, busy atm, will answer you asap
	SendInput, {Enter}
Return
```
